server/sonar-server/src/main/java/org/sonar/server/permission/index/NeedAuthorizationIndexer.java

```diff
@@ -17,38 +17,20 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-package org.sonar.server.issue.index;
+package org.sonar.server.permission.index;
 
-import com.google.common.collect.Maps;
-import org.sonar.server.es.BaseDoc;
-
-public class IssueAuthorizationDoc extends BaseDoc {
-
-  public IssueAuthorizationDoc() {
-    super(Maps.newHashMap());
-  }
-
-  @Override
-  public String getId() {
-    return projectUuid();
-  }
-
-  @Override
-  public String getRouting() {
-    return projectUuid();
-  }
-
-  @Override
-  public String getParent() {
-    return null;
-  }
+/**
+ * An {@link NeedAuthorizationIndexer} defines how
+ * a {@link org.sonar.server.es.ProjectIndexer} populates
+ * the type named {@link AuthorizationTypeSupport#TYPE_AUTHORIZATION}, which
+ * is used to verify that a user can access to projects.
+ */
+public interface NeedAuthorizationIndexer {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Annotate the "NeedAuthorizationIndexer" interface with the @FunctionalInterface annotation [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1609)